mobx_codegen/lib/src/mobx_codegen_base.dart

```diff
@@ -20,55 +20,87 @@ import 'package:mobx_codegen/src/template/util.dart';
 import 'package:source_gen/source_gen.dart';
 
 class StoreGenerator extends Generator {
-  final _storeChecker = const TypeChecker.fromRuntime(Store);
+  final _storeMixinChecker = const TypeChecker.fromRuntime(Store);
+  final _storeAnnotationChecker = const TypeChecker.fromRuntime(MakeStore);
 
   @override
   FutureOr<String> generate(LibraryReader library, BuildStep buildStep) {
-    Iterable<String> generate(ClassElement baseClass) sync* {
-      final mixedClass =
-          library.classes.where((c) => c != baseClass).firstWhere((c) {
-        if (c.supertype.typeArguments.isNotEmpty &&
-            baseClass.typeParameters.length ==
-                c.supertype.typeArguments.length) {
-          final t = baseClass.type.instantiate(c.supertype.typeArguments);
-          return t.isSupertypeOf(c.type);
-        }
-        return c.supertype == baseClass.type;
-      }, orElse: () => null);
-
-      if (mixedClass != null) {
-        yield generateStoreClassCode(library, baseClass, mixedClass);
+    final generatedMixins = library.classes
+        .where(isValidMixinStoreClass)
+        .expand((c) => generateCodeForMixinStore(library, c));
+    final generatedSubclasses = library.classes
```

Comment:
We need to handle the case where both the `@store` and `Store` mixin are used together

Reply:
Can you explain more?

Also, do I have to be making sure the user's class implements (or mixes in) Store?

Reply:
We have to treat the `@store` and `Store` mixin as mutually exclusive. However it is possible that someone uses them both together, in which case we should warn them and error out on using only one of the two.

Reply:
Also when you use the `@store` annotation, there is no need to use the mixin as we are generating a the public class directly

Reply:
Cool. Got it.

Just out of curiosity, the `Store` class' `dispose()` method. Does anything in the framework call that?

Reply:
No one is calling it. It was added an example to show that you could add such methods and override them in the outer public class.